Content.Server/Changeling/ChangelingSystem.cs

```diff
@@ -0,0 +1,1114 @@
+    {
+        if (!TryUseAbility(uid, comp, args))
+            return;
+
+        var reagents = new List<(string, FixedPoint2)>()
+        {
+            ("Synaptizine", 5f),
+        };
+        if (TryInjectReagents(uid, reagents))
+            _popup.PopupEntity(Loc.GetString("changeling-inject"), uid, uid);
+        else
+        {
+            _popup.PopupEntity(Loc.GetString("changeling-inject-fail"), uid, uid);
+            return;
+        }
+        PlayMeatySound(uid, comp);
+    }
+    // john space made me do this
+    public void OnHealUltraSwag(EntityUid uid, ChangelingComponent comp, ref ActionFleshmendEvent args)
+    {
+        if (!TryUseAbility(uid, comp, args))
+            return;
+
+        var reagents = new List<(string, FixedPoint2)>()
+        {
+            ("Impedrezene", 2.5f),
+            ("Ichor", 15f),
+            ("TranexamicAcid", 5f)
+        };
+        if (TryInjectReagents(uid, reagents))
+            _popup.PopupEntity(Loc.GetString("changeling-fleshmend"), uid, uid);
+        else return;
+        PlayMeatySound(uid, comp);
+    }
+    public void OnLastResort(EntityUid uid, ChangelingComponent comp, ref ActionLastResortEvent args)
+    {
+        if (!TryUseAbility(uid, comp, args))
+            return;
+
+        // todo: implement
+    }
+    public void OnLesserForm(EntityUid uid, ChangelingComponent comp, ref ActionLesserFormEvent args)
+    {
+        if (!TryUseAbility(uid, comp, args))
+            return;
+
+        var newUid = TransformEntity(uid, protoId: "MobMonkey", comp: comp);
+        if (newUid == null)
+        {
+            comp.Chemicals += Comp<ChangelingActionComponent>(args.Action).ChemicalCost;
+            return;
+        }
+
+        PlayMeatySound((EntityUid) newUid, comp);
+        var loc = Loc.GetString("changeling-transform-others", ("user", Identity.Entity((EntityUid) newUid, EntityManager)));
+        _popup.PopupEntity(loc, (EntityUid) newUid, PopupType.LargeCaution);
+
+        comp.IsInLesserForm = true;
+    }
+    public void OnSpacesuit(EntityUid uid, ChangelingComponent comp, ref ActionSpacesuitEvent args)
+    {
+        if (!TryUseAbility(uid, comp, args))
+            return;
+
+        if (!TryToggleItem(uid, SpacesuitPrototype, ref comp.SpacesuitEntity, "outerClothing")
+        || !TryToggleItem(uid, SpacesuitHelmetPrototype, ref comp.SpacesuitHelmetEntity, "head"))
+        {
+            _popup.PopupEntity(Loc.GetString("changeling-equip-armor-fail"), uid, uid);
+            return;
+        }
+
+        if (comp.SpacesuitEntity != null)
+            _popup.PopupEntity(Loc.GetString("changeling-equip-spacesuit-start"), uid, uid);
+        else
+            _popup.PopupEntity(Loc.GetString("changeling-equip-end"), uid, uid);
+
+        PlayMeatySound(uid, comp);
+    }
+    public void OnHivemindAccess(EntityUid uid, ChangelingComponent comp, ref ActionHivemindAccessEvent args)
```

Comment:
Shouldn't it be a passive effect that activates right after bying?

Reply:
iunno how to make such 😅